### ISSUE TITLE
Mac 0.5.1 fails to parse host that begin with a number if port is specified

### ISSUE THREAD

**Issue by ebriney** [docker/docker-credential-helpers#71]
```
docker login 8wrewr:1516
error getting credentials - err: exit status 1, out: `parse 8wrewr:1516: first path segment in URL cannot contain colon`
```

```
docker login 10.25.10.36:5000
error getting credentials - err: exit status 1, out: `parse 10.25.10.36:5000: first path segment in URL cannot contain colon`
```

NB: windows version is working, I didn't test the linux one

**Comment by n4ss** [comment]
https://github.com/docker/docker-credential-helpers/pull/55 solves that I think as we prepend a "https" to the url before the `url.Parse` call.

/cc @thaJeztah

**Comment by gtardif** [comment]
Any news regarding this one ? 
We'd like to include a fix in D4Mac 17.06.0-rc3

**Comment by thaJeztah** [comment]
Same issue as is being resolved in https://github.com/docker/docker-credential-helpers/pull/55, but there was some discussion around the implementation. I had an alternative implementation for that that I shared with @n4ss, but we should decide what we go for;

```go
func splitServer(serverURL string) (*C.struct_Server, error) {
    if !regexp.MustCompile(`^([a-zA-Z][-+.a-zA-Z0-9]+:)?//`).MatchString(serverURL) {
        serverURL = "//" + serverURL
    }
    u, err := url.Parse(serverURL)
    if err != nil {
        return nil, err
    }
    if u.Scheme != "" && u.Scheme != "https" && u.Scheme != "http" {
        return nil, errors.New("unsupported scheme: " + u.Scheme)
    }
    if url.Host() == "" {
        return nil, errors.New("no hostname in URL")
    }
    proto := C.kSecProtocolTypeHTTPS
    if u.Scheme == "http" {
        proto = C.kSecProtocolTypeHTTP
    }
    var port int
    if u.Port() != "" {
        port, err := strconv.Atoi(u.Port())
        if err != nil {
            return nil, err
        }
    }

    return &C.struct_Server{
        proto: C.SecProtocolType(proto),
        host:  C.CString(u.Host()),
        port:  C.uint(port),
        path:  C.CString(u.Path()),
    }, nil
}
```

**Comment by thaJeztah** [comment]
Oh, I just notice @n4ss mentioned #55 already 😅

**Comment by thaJeztah** [comment]
This is fixed in version 0.5.2 through https://github.com/docker/docker-credential-helpers/issues/72 👍
